src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -470,6 +470,8 @@ export class HttpConnection implements IConnection {
 
         if (this._connectionState === ConnectionState.Disconnected) {
             this._logger.log(LogLevel.Debug, `Call to HttpConnection.stopConnection(${error}) was ignored because the connection is already in the disconnected state.`);
+            // Make sure HttpConnection.stop() won't be blocked on the promise
+            this._stopPromiseResolver();
```

Comment:
At this point, start() has already failed and transitioned the connection to ConnectionState.Disconnected which means the connection could theoretically be restarted and stopped by the time this is run causing this to resolve a subsequent call to stop() incorrectly. I think we should remove the following call to `this._stopConnection()` while keeping the log for diagnostic purposes:

https://github.com/dotnet/aspnetcore/blob/614908248d20c4a1572d1a3bbc7a997d12be9d01/src/SignalR/clients/ts/signalr/src/HttpConnection.ts#L216-L219

and just call `this._stopPromiseResolver();` in the `_startInternal()` catch before returning the rejected promise:

https://github.com/dotnet/aspnetcore/blob/614908248d20c4a1572d1a3bbc7a997d12be9d01/src/SignalR/clients/ts/signalr/src/HttpConnection.ts#L294-L299

Reply:
>the connection could theoretically be restarted and stopped by the time this is run causing this to resolve a subsequent call to stop() incorrectly.

I don't think it can, but even if it could, the connection is in the disconnected state already so calling this in the worst case will do nothing.

> just call `this._stopPromiseResolver();` in the `_startInternal()` catch before returning the rejected promise

That's probably fine. Have to check for undefined, but that's ok.

> I think we should remove the following call to `this._stopConnection()` while keeping the log for diagnostic purposes

Hmm interesting, my biggest concern with that is `_stopConnection` cleans up some fields, but I don't think any of them will be set if the transport isn't set. And it gets rid of some extraneous logs if you hit that code path :+1:

Reply:
> so calling this in the worst case will do nothing.

I don't think so. Like I said

>  the connection could theoretically be restarted and stopped by the time this is run causing this to resolve a subsequent call to stop() incorrectly.

Reply:
Talked offline. This is important not for when the reconnect delay is canceled but for when a reconnect attempt fails simultaneously with a call to HubConnection.stop().